src/Libraries/DesignScriptBuiltin/Builtin.cs

```diff
@@ -53,6 +53,11 @@ public static object ValueAtIndexInForLoop(IList list, int index)
                 return ValueAtIndex(list, index);
             }
 
+            public static object ValueAtIndexInForLoop(string stringList, int index)
```

Comment:
what code triggers this?

Reply:
A `for` loop is now compiled to an AST containing this method. In other words this method is triggered when we run `for` loops.

Reply:
More specifically it is triggered when we iterate over strings such as here:
```
r = [Imperative]
{
    s = "foo";
    for (x in "bar")
    {
         s  = s + x;
    }
    return = s;
};
```
Expected result: "foobar"